src/esockd_acceptor_sup.erl

```diff
@@ -139,12 +160,27 @@ tune_socket(Sock, [{tune_buffer, true}|More]) ->
     case esockd_transport:getopts(Sock, [sndbuf, recbuf, buffer]) of
         {ok, BufSizes} ->
             BufSz = lists:max([Sz || {_Opt, Sz} <- BufSizes]),
-            _ = esockd_transport:setopts(Sock, [{buffer, BufSz}]),
+            case esockd_transport:setopts(Sock, [{buffer, BufSz}]) of
+                ok ->
+                    tune_socket(Sock, More);
+                Error ->
+                    Error
+            end;
+        Error ->
+            Error
+    end;
+tune_socket(Sock, [fast_check | More]) ->
+    %% by default, for gen_tcp sockets (port)
+    %% try to getopts so to detect einval and enotconn sonner
+    %% (before spawn a connection process)
+    %% for other types of sockets, errors will be detected
+    %% by the upgrade functions anyway
+    case esockd_transport:getopts(Sock, [active]) of
```

Comment:
unsue if worth to do it here by using getopts, the overhead may slow down the shuffling of accept queue.

Reply:
Had a look again, socket upgrade is actually done by the owner process.
fast_check has been removed.